specification/redis/resource-manager/Microsoft.Cache/stable/2022-06-01/redis.json

```diff
@@ -2670,12 +2670,12 @@
         "geoReplicatedPrimaryHostName": {
           "type": "string",
           "readOnly": true,
-          "description": "Fully qualified geo hostname which points to geo-primary cache among the linked redis caches."
+          "description": "DNS which will always point to current geo-primary cache among the linked redis caches so that Geo failover support can be enabled."
         },
         "primaryHostName": {
           "type": "string",
           "readOnly": true,
-          "description": "DNS corresponding to geo-primary cache among the linked redis caches."
+          "description": "DNS name that resolves to the current geo-primary cache among the linked redis caches."
```

Comment:
I feel like I understand both descriptions now but... I also suspect I am probably missing something. aren't they the saying the same thing? What's the difference? They are both the current geo-primary??

Reply:
Correct, both are current geo-primary,  **geoReplicatedPrimaryHostName** will change after failover when geo secondary becomes new geo primary. Similarly, **primaryHostName** will always point to current geo-primary cache DNS. 
For detailed architecture, you can refer this Design Doc: [1 Click Failover](https://microsoft.sharepoint.com/:w:/t/AzureCache/EWqtXw528zNGoi50HO6oL4IBcY0WoCaeep6b1NkTpzwlOw?e=rFieI2)

Reply:
Ah, so the second one, primaryHostName is more like 'The hostname of the current geo-primary cache among the linked redis caches. During a swap of geo-replication roles, the value of primaryHostName will change, because it is updated to the hostname of the new geo-primary'. 

And the first one, geoReplicatedPrimaryHostName is more like 'Unchanging DNS name. This DNS name never appears to change, but it will always dynamically resolve to the current geo-primary cache among the linked redis caches, even after a failover when the linked servers swap their geo-replication roles (primary <-> secondary). Applications can use this value in their configuration to automatically switch between regions.'